packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -317,7 +318,24 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
         }
 
         if (delayMs > 0) {
-            await delay(delayMs);
+            let timer;
+            let resolveOpPromiseFn;
+            // Create a listener that will break the delay if we've exceeded the initial delay ops count.
+            const opsListenerFn = () => {
+                if (this.summaryCollection.opsSinceLastAck >= this.opsToBypassInitialDelay) {
+                    clearTimeout(timer);
+                    resolveOpPromiseFn();
+                }
+            };
+            // Create a Promise that will resolve when the delay expires.
+            const delayPromise = new Promise<void>((resolve) => {
+                timer = setTimeout(() => resolve(), delayMs);
+            });
+            // Create a Promise that will resolve if the ops count passes the threshold.
+            const opPromise = new Promise<void>((resolve) => { resolveOpPromiseFn = resolve; });
+            this.summaryCollection.addOpListener(opsListenerFn);
```

Comment:
SummaryCollection already has an op listener for delta manager ops and emits them. Consider using the events emitted by SummaryCollection instead of adding new listeners.
Maybe something like - `this.summaryCollection.on(OpActionEventName, ())`?

Reply:
Thanks. I'll try that.

Reply:
What I'm doing here, though, is adding an action that we'll take on a delta manager "op" event only while we're waiting for these promises to resolve.

Reply:
Oh, but you're talking about keying on the events that are fired as a result of calling SummaryCollection.handleOp. Yeah, I like that better.

Reply:
Okay, I remember why I did it like I did. I wasn't able to use OpActionEventName directly in this way. It was easier to listen for the deltamanager "op" event than for all of the events that SummaryCollection might emit under handleOp.